src/track/globaltrackcache.cpp

```diff
@@ -313,38 +354,19 @@ bool GlobalTrackCache::resolveInternal(
                     << "Resolving track by id"
                     << trackId;
         }
-        const auto trackById(m_tracksById.find(trackId));
-        if (m_tracksById.end() != trackById) {
-            // Cache hit
-            TrackRef resolvedTrackRef((*trackById).ref);
-            TrackPointer pResolvedTrack((*trackById).weakPtr);
-            if (pResolvedTrack) {
-                if (kLogger.debugEnabled()) {
-                    kLogger.debug()
-                            << "Cache hit - found track by id"
-                            << resolvedTrackRef;
-                }
-                *pCacheResolver = GlobalTrackCacheResolver(
-                        std::move(*pCacheResolver),
-                        GlobalTrackCacheLookupResult::HIT,
-                        resolvedTrackRef,
-                        pResolvedTrack);
-                return true;
-            }
-            // We don't expect that this might ever happen, but let's even
-            // handle this special case!
-            VERIFY_OR_DEBUG_ASSERT(pResolvedTrack) {
-                // Explicitly evict the cached track before the deleter does it
-                kLogger.warning()
-                        << "Cache hit - evicting zombie track"
-                        << resolvedTrackRef;
-                // The cache must stay locked after evicting the zombie entry
-                // so we don't pass the locker on!
-                Track* pEvictedTrack = evictInternal(nullptr, resolvedTrackRef);
-                DEBUG_ASSERT((nullptr == pEvictedTrack) ||
-                        (pEvictedTrack == (*trackById).plainPtr));
-                // ...and continue like it has not been found
+        auto trackRefPointer = lookupInternal(trackId);
```

Comment:
IMHO the ISO CPP guideline to use std::pair for two value return function is a mistake. A named struct would be a better choice. 
I like the "out" keyword in c# for output parameters as well. We can consider to model it via a "out<T>"  decoration template.   
Do you know if std::pair or a struct has any performance drawback compared to a pointer parameter as second return value?

Reply:
I just needed a tuple with 2 elements to pass those internal results around in order to avoid redundant code. The 2 values belong together and shouldn't be torn apart into obscure output parameters. Creating a new type that is not used anywhere else is tedious and not really worth the effort. Unfortunately crufty, old C++ has no first class support for tuples.

Well, maybe I can write a dedicated TrackRefPointer type. But writing so much unnecessary code that needs top be maintained is one of the reasons why I feel uncomfortable with C++ and want to move on ;)